sdk/src/types/block/output/foundry.rs

```diff
@@ -32,6 +33,54 @@ use crate::types::{
     ValidationParams,
 };
 
+impl_id!(pub FoundryId, 38, "Defines the unique identifier of a foundry.");
+
+#[cfg(feature = "serde")]
+string_serde_impl!(FoundryId);
+
+impl From<TokenId> for FoundryId {
+    fn from(token_id: TokenId) -> Self {
+        Self::new(*token_id)
+    }
+}
+
+impl FoundryId {
+    /// Builds a new [`FoundryId`] from its components.
+    pub fn build(account_address: &AccountAddress, serial_number: u32, token_scheme_kind: u8) -> Self {
+        let mut bytes = [0u8; Self::LENGTH];
+        let mut packer = SlicePacker::new(&mut bytes);
+
+        // PANIC: packing to an array of the correct length can't fail.
+        Address::Account(*account_address).pack(&mut packer).unwrap();
+        serial_number.pack(&mut packer).unwrap();
+        token_scheme_kind.pack(&mut packer).unwrap();
+
+        Self::new(bytes)
+    }
+
+    /// Returns the [`AccountAddress`] of the [`FoundryId`].
+    pub fn account_address(&self) -> AccountAddress {
+        // PANIC: the lengths are known.
+        AccountAddress::from(AccountId::new(self.0[1..AccountId::LENGTH + 1].try_into().unwrap()))
```

Comment:
It's a bit unfortunate that `FoundryId` has to have exact knowledge about how `Address` packs its stuff (with 1 byte for the `AddressType` I assume). I know it's annoying but instead of the magic number `1` something like an `Address::OFFSET` constant would make sense.

Reply:
Honestly I wouldn't add a constant just for that specific use case. I can throw in a size_of if you prefer but it doesn't change much 🤷🏻‍♂️